### PR TITLE
Issue #1670 - Fix date in link urls of search response

### PR DESCRIPTION
**Pull request by tbieste**
Fixes dates in link urls of search response to represent the actual date specified in the search request.

Signed-off-by: Troy Biesterfeld <tbieste@us.ibm.com>

### CODE REVIEW

File: fhir-search/src/test/java/com/ibm/fhir/search/test/QueryParameterValueTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+    @Test
+    public void testValueString() throws Exception {
+        QueryParameterValue testObj = new QueryParameterValue();
+        String valueString = "valueString";
+        testObj.setValueString(valueString);
+        assertEquals(testObj.getValueString(), valueString);
+        assertEquals(testObj.toString(), "valueString");
+        assertEquals(testObj.toString(), "valueString");
```
Comment (michaelwschroeder) [inline fhir-search/src/test/java/com/ibm/fhir/search/test/QueryParameterValueTest.java:32]: why the duplicate lines for toString() assertion on each test?

**Review by michaelwschroeder — APPROVED** [review]

**Comment by michaelwschroeder** [comment]
LGTM - just one minor comment

File: fhir-search/src/test/java/com/ibm/fhir/search/test/QueryParameterValueTest.java (hunk 1 above)
Comment (tbieste) [reply fhir-search/src/test/java/com/ibm/fhir/search/test/QueryParameterValueTest.java:32]: This was to test a fix to a previous bug where the "delim" member variable kept state across toString calls, so the 2nd call would not match the 1st call.

Comment (michaelwschroeder) [reply fhir-search/src/test/java/com/ibm/fhir/search/test/QueryParameterValueTest.java:32]: got it - thanks!

File: fhir-search/src/test/java/com/ibm/fhir/search/date/DateTimeHandlerTest.java [hunk 2]
```diff
@@ -579,7 +580,7 @@ public void testYearMonthDaySeparatorHourMinutesSecondsMicrosecondsParserZoneBad
     @Test(expectedExceptions = FHIRSearchException.class)
     public void testYearMonthDaySeparatorHourMinutesSecondsMicrosecondsParserZoneBadZoneUseWithDay()
             throws FHIRSearchException {
-        // This case is 
+        // This case is
```
Comment (lmsurpre) [inline fhir-search/src/test/java/com/ibm/fhir/search/date/DateTimeHandlerTest.java:583]: probably better just deleting this one :-)
```suggestion
```

**Review by lmsurpre — APPROVED** [review]
LGTM

File: fhir-search/src/test/java/com/ibm/fhir/search/date/DateTimeHandlerTest.java (hunk 2 above)
Comment (tbieste) [reply fhir-search/src/test/java/com/ibm/fhir/search/date/DateTimeHandlerTest.java]: Removed.

**Comment by tbieste** [comment]
Comments addressed.  Ready to merge.

**Pull request merged**